### PR TITLE
Add M3U Radio music provider

### PR DESCRIPTION
**Pull request by cladkins**
# What does this implement/fix?

Adds a new music provider, **M3U Radio** (`m3uradio`), which imports radio stations from a self-hosted M3U / IPTV playlist URL and exposes them as a native radio source: library sync, browse by `group-title`, and search.

Motivation: self-hosters maintain a station list in an M3U editor. Today the options are manually adding stations one by one via the builtin provider, or one-shot import scripts that go stale. This makes the playlist itself the source of truth.

Design notes:

- **Stable item IDs**: `tvg-id` when present, otherwise a 16-char sha1 of `name|url` — favorites and history survive playlist refreshes.
- **Refresh**: the playlist is re-fetched in a `sync_library` override on every scheduled library sync, so playlist edits flow in automatically.
- **Group filter**: optional comma-separated `group-title` include-filter, for mixed TV/radio IPTV playlists. `multi_instance: true` allows several playlists.
- **Streaming**: direct URL, `StreamType.HTTP`, no seeking (live radio). ICY metadata works out of the box.
- **No external dependencies** — the parser is stdlib-only and covered by unit tests (`tests/providers/m3uradio/`), including the comma-inside-quoted-attributes edge case.

Tested locally against dev: library sync (441-station real-world playlist), browse, search, and end-to-end playback with ICY now-playing metadata.

**Related issue (if applicable):**

- n/a

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue) — `bugfix`
- [ ] New feature (non-breaking change which adds functionality) — `new-feature`
- [ ] Enhancement to an existing feature — `enhancement`
- [x] New music/player/metadata/plugin provider — `new-provider`
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected) — `breaking-change`
- [ ] Refactor (no behaviour change) — `refactor`
- [ ] Documentation only — `documentation`
- [ ] Maintenance / chore — `maintenance`
- [ ] CI / workflow change — `ci`
- [ ] Dependencies bump — `dependencies`

## Checklist

- [x] The code change is tested and works locally.
- [x] `pre-commit run --all-files` passes.
- [x] `pytest` passes, and tests have been added/updated under `tests/` where applicable.
- [ ] For changes to shared models, the companion PR in `music-assistant/models` is linked. *(n/a — no model changes)*
- [ ] For changes affecting the UI, the companion PR in `music-assistant/frontend` is linked. *(n/a — no UI changes)*
- [X] I have read and complied with the project's [AI Policy](https://github.com/music-assistant/.github/blob/main/AI_POLICY.md) for any AI-assisted contributions.
- [ ] I have raised a PR against the documentation repository targeting the main or beta branch as appropriate. *(docs PR to follow)*

Generated with help from Claude Code

### CODE REVIEW

**Comment by cladkins** [comment]
Closing while I review locally — will reopen when ready for review.

File: music_assistant/providers/m3uradio/__init__.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+    async def handle_async_init(self) -> None:
+        """Handle async initialization of the provider."""
+        self._stations = {}
+        await self._refresh_stations()
+
+    async def sync_library(self, media_type: MediaType) -> None:
+        """Run library sync for this provider."""
+        # re-fetch the playlist so scheduled syncs pick up playlist changes
+        await self._refresh_stations()
```
Comment (marcelveldt) [inline music_assistant/providers/m3uradio/__init__.py:114]: why not do the refresh in the get_library_radios call and then drop this method ?

File: music_assistant/providers/m3uradio/__init__.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+                        path=st["logo"],
+                        provider=self.instance_id,
+                        remotely_accessible=True,
+                    )
+                ]
+            )
+        return radio
+
+
+def parse_m3u(content: str) -> list[dict[str, str]]:
```
Comment (marcelveldt) [inline music_assistant/providers/m3uradio/__init__.py:231]: we have an existing m3u parser in helpers, maybe use that ? If it's missing some fields we could extend it

File: music_assistant/providers/m3uradio/__init__.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        """Get radio station details."""
+        st = self._stations.get(prov_radio_id)
+        if not st:
+            await self._refresh_stations()
+            st = self._stations.get(prov_radio_id)
+        if not st:
+            raise MediaNotFoundError(f"Station {prov_radio_id} not found")
+        return self._parse_radio(st)
+
+    async def search(
```
Comment (marcelveldt) [inline music_assistant/providers/m3uradio/__init__.py:132]: do you really need search ? If all stations are imported to the library anyways, the library will already provide the search results.

File: music_assistant/providers/m3uradio/__init__.py (hunk 1 above)
Comment (cladkins) [reply music_assistant/providers/m3uradio/__init__.py:114]: Correcting this by moving _refresh_stations() to the top of get_library_radios()

File: music_assistant/providers/m3uradio/__init__.py (hunk 3 above)
Comment (cladkins) [reply music_assistant/providers/m3uradio/__init__.py:132]: Removed the search

File: music_assistant/providers/m3uradio/__init__.py [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+    instance_id: str | None = None,  # noqa: ARG001
+    action: str | None = None,  # noqa: ARG001
+    values: dict[str, ConfigValueType] | None = None,  # noqa: ARG001
+) -> tuple[ConfigEntry, ...]:
+    """Return config entries to set up this provider."""
+    return (
+        ConfigEntry(
+            key=CONF_M3U_URL,
+            type=ConfigEntryType.STRING,
+            label="M3U / IPTV playlist URL",
```
Comment (marcelveldt) [inline music_assistant/providers/m3uradio/__init__.py:83]: we recently migrated the backend to support localization (which was added right after your pr got opened). Can you adjust these hardcoded strings to translation_key and a strings.json in your provider folder ?

Look at some of the other providers how this is done.

File: music_assistant/providers/m3uradio/__init__.py (hunk 3 above)
Comment (marcelveldt) [reply music_assistant/providers/m3uradio/__init__.py:132]: Did you not yet push that change as I'm still seeing the search

Comment (cladkins) [reply music_assistant/providers/m3uradio/__init__.py:132]: I've been back up at work. I have not pushed the changes yet.

**Comment by OzGav** [comment]
Marking this as draft so we know what needs attention. Please press the Ready for Review button when it is ready again.

File: music_assistant/providers/m3uradio/__init__.py [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+def _hash_id(name: str, url: str) -> str:
+    return hashlib.sha1(f"{name}|{url}".encode()).hexdigest()[:16]
+
+
+def _slug(text: str) -> str:
+    return re.sub(r"[^a-z0-9]+", "-", text.lower()).strip("-") or "ungrouped"
+
+
+def _guess_content_type(url: str) -> ContentType:
```
Comment (OzGav) [inline music_assistant/providers/m3uradio/__init__.py:279]: `ContentType.try_parse()` is in the models already? More importantly, the established convention for live radio is to not guess, somafm sets `content_type=ContentType.UNKNOWN` and lets the stream be probed. You should drop the helper and its test and use` ContentType.UNKNOWN` (matching somafm) or `ContentType.try_parse(st["url"])`

**Comment by OzGav** [comment]
You don't override the base class `is_streaming_provider` property therefore it is set as TRUE

The practical effect is in the base `sync_library` deletion path with False, a station removed from the playlist gets fully removed from the library on the next sync but with True, it lingers as an orphaned `in_library=False` mapping. 

What is your intent?

File: music_assistant/providers/m3uradio/__init__.py [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+        """Fetch the playlist and rebuild the station index."""
+        url = self.config.get_value(CONF_M3U_URL)
+        if not url:
+            self.logger.warning("No M3U URL configured")
+            self._stations = {}
+            return
+        groups_raw = self.config.get_value(CONF_GROUPS) or ""
+        group_filter = {g.strip().lower() for g in str(groups_raw).split(",") if g.strip()}
+        try:
+            async with self.mass.http_session.get(str(url)) as resp:
```
Comment (OzGav) [inline music_assistant/providers/m3uradio/__init__.py:190]: You should set a timeout value here

File: music_assistant/providers/m3uradio/__init__.py [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+                resp.raise_for_status()
+                content = await resp.text()
```
Comment (OzGav) [inline music_assistant/providers/m3uradio/__init__.py:192]: This could raise an uncaught error. You could widen the caught errors below, but elsewhere in the codebase `detect_charset` is used to handle missing/wrong server charsets so read the raw bytes and decode with `errors="replace"` so it can't throw (see `fetch_playlist` in [helpers/playlists.py](https://github.com/music-assistant/server/blob/claude/intelligent-thompson-kpeqpw/helpers/playlists.py)).

**Comment by OzGav** [comment]
@cladkins any update on this one?

**Comment by OzGav** [comment]
We will close this soon due to no progress.
